pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -178,13 +181,17 @@ assert withImportd -> withCompression;
 assert withCoredump -> withCompression;
 assert withHomed -> withCryptsetup;
 assert withHomed -> withPam;
+assert withHomed -> withOpenSSL;
+assert withFido2 -> withOpenSSL;
+assert withSysupdate -> withOpenSSL;
+assert withImportd -> (withGcrypt || withOpenSSL);
 assert withUkify -> (withEfi && withBootloader);
 assert withRepart -> withCryptsetup;
 assert withBootloader -> withEfi;
 
 let
   wantCurl = withRemote || withImportd;
-  wantGcrypt = withResolved || withImportd;
```

Comment:
Should also add  a  `withResolved -> withGcrypt` assertion?

Reply:
see the comment below, sorry my internet was down, so the comment i wanted to add, did take some time :slightly_smiling_face:

Reply:
> Are withGcrypt = true and withOpenssl = true compatible?

explicit yes, as that was also already possible on the old setup (withFido did include openssl and resolved enabled gcrypt, which are both active on default)

Reply:
If it can build without it, I think it's fine to leave it out. Assertions are added to make it easier to debug build failures (rather than attempt to build and have the user dig through the logs to find out their options aren't compatible).